cumulus_library/databases.py

```diff
@@ -64,36 +66,33 @@ def _parse_found_schema(self, expected: dict[dict[list]], schema: list[list]):
         is a first pass, ignoring complexities of differing database variable
         types, just iterating through looking for column names.
 
-
-        TODO: on a per database instance, consider a more nuanced approach
-        if needed
+        TODO: on a per database instance, consider a more nuanced approach if needed
+              (compared to just checking if the schema contains the field name)
         """
         output = {}
-        for column, _ in expected.items():
-            output[column] = {}
-            if col_schema := schema[column.lower()]:
-                # is this an object column?
-                if len(expected[column]) > 0:
-                    for field in expected[column]:
-                        col_schema = col_schema.split(field, 1)
-                        if len(col_schema) != 2:
-                            output[column][field] = False
-                            col_schema = col_schema[0]
-                        else:
-                            output[column][field] = True
-                            col_schema = col_schema[1]
```

Comment:
This here was the source of one of the bugs - after splitting on each field, future scans would only look at the part *after* the split. So order of fields matters in a way that is hard to predict.

I've switched from splitting entirely and just search in the entire column schema for each field.

Reply:
So, this does create a different kind of bug, though maybe note as critical as the found one: for columns where a object parameter occurs more than once, this may over-identify data as being present. specifically, id shows up in a lot of places. 

if this :does: hit, it will prevent a table from being run, which is the spirit of the previous version - take a few false negatives for more true positives.

If this is blocking you now, that's fine, but we might need to more robustly solve this in the near future.

Reply:
We chatted about this in person - I think we agree it is maybe not a new bug but just another instance of the existing flakiness of "searching in a big string for a field". We need to move to per-db-backend schema parsing to fix this and to allow deeper-depth schema searching.